gems/aws-sdk-core/lib/aws-sdk-core/binary/event_parser.rb

```diff
@@ -0,0 +1,92 @@
+module Aws
+  module Binary
+    # @api private
+    class EventParser
+
+      include Seahorse::Model::Shapes
+
+      # @param [Class] parser_class
+      # @param [Seahorse::Model::ShapeRef] rules
+      def initialize(parser_class, rules)
+        @parser_class = parser_class
+        @rules = rules
+      end
+
+      # Parse raw event message into event struct
+      # based on its ShapeRef
+      #
+      # @return [Struct] Event Struct
+      def apply(raw_event)
+        parse(raw_event)
+      end
+
+      private
+
+      def parse(raw_event)
+        message_type = raw_event.headers.delete(":message-type")
+        if message_type
+          case message_type.value
+          when 'error'
+            parse_error_event(raw_event)
+          when 'event'
+            parse_event(raw_event)
+          else
+            raise 'Unrecognized :message-type value for the event'
+          end
+        else
+          # no :message-type header, regular event by default
+          parse_event(raw_event)
+        end
+      end
+
+      def parse_error_event(raw_event)
+        error_code = raw_event.headers.delete(":error-code")
+        error_message = raw_event.headers.delete(":error-message")
+        Aws::EventError.new(
+          :error,
+          error_code ? error_code.value : error_code,
+          error_message ? error_message.value : error_message
+        )
+      end
+
+      def parse_event(raw_event)
+        event_type = raw_event.headers.delete(":event-type").value
+        # content_type = raw_event.headers.delete(":content-type").value
+
+        # TODO Pending
```

Comment:
?

Reply:
Yeah ... though current no real server response is available for testing this, I'll adding this